### PR TITLE
New Features: Docker integration and using ENV for debug mode.

### PR DESCRIPTION
**Pull request by stowwe**
# Changes
- Docker support
    - Added [Dockerfile](./Dockerfile) - Runs the webapp version of the project. 
    - Added [docker-compose.yml](./docker-compose.yml), contains a basic compose build with parameters.
    - Added [docker.sh](./docker.sh), which builds/deploys the compose file in one go.
    - Added documentation to the [README](./README.md) for running the docker components. Also fixed a small typo!
- Webapp changes
    - Changed [web/app.js](web/app.js) to use the environment variable `DEBUG_MODE` instead of being hard-coded on.
- [requirements.txt](./requirements.txt) changes
    - Added flask. The standalone python app doesn't need it, but the webapp does.

Open to pulls/suggestions.

### CODE REVIEW

**Comment by TLMcNulty** [comment]
This looks good, thanks for contributing!

Unfortunately I don't have a computer I can test docker features with me at the moment but I have no problems with this change, it looks good. 

The only real change I can request is moving pip to the requirements.txt install, and I believe that was contributed after/at the same time as you put this together. 

Could I also request an addition to README.md covering "Running this project with docker." I know it's a simple configuration but I'd still like to have it documented.

**Comment by stowwe** [comment]
Should be all set now @YanFett, hope this helps! :smile:

File: README.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # Wintergatan Timing Data Analysis
 
-The community has long since clamoured for automatic peak detection and generation of timing data for the Wintergata Marble Machine project. An offhand comment during the 2023 Community Meetup colalesced into this project.
```
Comment (TLMcNulty) [inline README.md:3]: Great typo catch thanks!

File: README.md [hunk 2]
```diff
@@ -60,6 +60,38 @@ To install these libraries, run the following command:
 pip install -r requirements.txt
 ```
 
+### Running with Docker
```
Comment (TLMcNulty) [inline README.md:63]: Perfect update!

**Review by TLMcNulty — APPROVED** [review]
We run this in gunicorn on the web so important to note we'd need to build a different image if we wanted to deploy this way. Great work though your contributions are much appreciated!

**Pull request merged**